wren-ai-service/tools/providers-setup/ui_components.py

```
@@ -0,0 +1,551 @@
+                    selected_llm = st.selectbox(
+                        "LLM Model",
+                        options=pipeline_llm_options,
+                        index=pipeline_llm_options.index(value) if value in pipeline_llm_options else 0,
+                        key=f"llm_{original_idx}"
+                    )
+                else:
+                    st.markdown(f"**{key}:** `{value}`")
+
+            if st.button("💾  Save this llm", key=f"save_{pipe_name}_{original_idx}"):
+                # ✅ use original index to update llm
+                st.session_state[ConfigState.PIPELINE_KEY]["pipes"][original_idx]["llm"] = selected_llm
```

Pylint (3.3.7)

[error] Possibly using variable 'selected_llm' before assignment

(E0606)